openl3/core.py

```diff
@@ -209,15 +214,327 @@ def process_file(filepath, output_dir=None, suffix=None, model=None,
+    model : keras.models.Model or None
+        Loaded model object. If a model is provided, then `input_repr`,
+        `content_type`, and `embedding_size` will be ignored.
+        If None is provided, the model will be loaded using
+        the provided values of `input_repr`, `content_type` and
+        `embedding_size`.
+    input_repr : "linear", "mel128", or "mel256"
+        Spectrogram representation used for to train audio part of audio-visual
+        correspondence model. Ignored if `model` is a valid Keras model.
+    content_type : "music" or "env"
+        Type of content used to train embedding. Ignored if `model` is
+        a valid Keras model.
+    embedding_size : 8192 or 512
+        Embedding dimensionality. Ignored if `model` is a valid
+        Keras model.
+    verbose : 0 or 1
+        Keras verbosity.
+
+    Returns
+    -------
+        embedding : np.ndarray [shape=(N, D)]
+            Array of embeddings for each frame.
+        timestamps : np.ndarray [shape=(N,)]
+            Array of timestamps for each frame. If `frame_rate` is None,
+            this is not returned.
+    """
+    if image.size == 0:
+        raise OpenL3Error('Got empty image')
+
+    # Warn user if image is all zero
+    if np.all(image == 0):
+        warnings.warn('Provided image is all zeros', OpenL3Warning)
+
+    if model is not None and not isinstance(model, keras.models.Model):
+        raise OpenL3Error('Invalid model provided. Must be of type keras.model.Models'
+                          ' but got {}'.format(str(type(model))))
+
+    if str(input_repr) not in ("linear", "mel128", "mel256"):
+        raise OpenL3Error('Invalid input representation "{}"'.format(input_repr))
+
+    if str(content_type) not in ("music", "env"):
+        raise OpenL3Error('Invalid content type "{}"'.format(content_type))
+
+    if embedding_size not in (8192, 512):
+        raise OpenL3Error('Invalid content type "{}"'.format(embedding_size))
+
+    if verbose not in (0, 1):
+        raise OpenL3Error('Invalid verbosity level {}'.format(verbose))
+
+    if (frame_rate is not None) and (not isinstance(frame_rate, Real) or frame_rate <= 0):
+        raise OpenL3Error('Invalid frame rate {}'.format(frame_rate))
+
+    # Check image array dimension
+    if image.ndim not in (3, 4):
+        raise OpenL3Error('RGB image array can only be 3D or 4D (sequence of videos)')
+
+    # Get embedding model
+    if model is None:
+        model = load_image_embedding_model(input_repr, content_type, embedding_size)
+
+    if image.shape[-1] != 3:
+        raise OpenL3Error('Need 3 channel images corresponding to RGB.')
+
+    # Preprocess image to scale appropriate scale
+    x = _preprocess_image_batch(image)
+
+    # Get embedding and timestamps
+    embedding = model.predict(x, verbose=verbose)
+
+    if frame_rate is not None:
+        ts = np.arange(embedding.shape[0]) / float(frame_rate)
+        return embedding, ts
+    else:
+        return embedding
```

Comment:
Isn't this a risky paradigm (where the expected number of output variables is dynamic)? Would it not be better to always return `embedding, ts`, and just set `ts` to `None` if a frame rate isn't provided?